csi/moac/pool.ts

```diff
@@ -129,7 +129,9 @@ export class Pool {
   // @param {object} replica      New replica object.
   //
   registerReplica(replica: Replica) {
-    assert(!this.replicas.find((r) => r.uuid === replica.uuid));
+    if (this.replicas.find((r) => r.uuid === replica.uuid)) {
+      this.unregisterReplica(replica)
+    }
```

Comment:
Is there a particular reason for handling this gracefully? Adding a replica that already exists to a pool could be a programming mistake and in that case, assert would be a better option for dealing with that.

Reply:
I noticed that this can happen when a node/mayastor is brought up and moac tries to recreate the replica. Seems like we add the replica through the response of create replica but we also pick it up again from the events?